### PR TITLE
fix: Apply new discount code routing

### PR DESCRIPTION
**Pull request by anhanh11001**
Fixes: #2163

### CODE REVIEW

**Comment by anhanh11001** [comment]
@liveHarshit not yet, I just change based on server PR. Do you already have an event with discount code or should I make a new one?

**Comment by liveHarshit** [comment]
> @liveHarshit not yet, I just change based on server PR. Do you already have an event with discount code or should I make a new one?

Yes but it's outdated. You need to create a new one.

**Comment by liveHarshit** [comment]
Have you tested it, or I can test it?

**Comment by anhanh11001** [comment]
Yes, it is not working. @liveHarshit 

@prateekj117 is this a correct endpoint:
https://api.eventyay.com/event/954/discount-code/DISS7

I have followed the change in your PR but this throw 404 error as requested URL not found on server. Or is it only updated in the debug mode?

File: app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt [hunk 1]
```diff
@@ -7,6 +7,10 @@ import retrofit2.http.Query
 
 interface DiscountApi {
 
-    @GET("discount-codes/{code}?include=event,tickets")
-    fun getDiscountCodes(@Path("code") code: String, @Query("filter") filter: String): Single<DiscountCode>
+    @GET("event/{eventId}/discount-code/{code}?include=event,tickets")
```
Comment (prateekj117) [inline app/src/main/java/org/fossasia/openevent/general/discount/DiscountApi.kt]: It's `/events/{eventId}/discount-codes/{code}`.

**Comment by prateekj117** [comment]
@anhanh11001 No, it's wrong. Correct endpoint would be: `https://api.eventyay.com/events/954/discount-codes/DISS7`.

**Comment by iamareebjamal** [comment]
https://api.eventyay.com/v1/events/954/discount-codes/DISS7

**Comment by anhanh11001** [comment]
It's working now

File: app/build.gradle [hunk 2]
```diff
@@ -63,7 +63,9 @@ android {
                 signingConfig signingConfigs.release
         }
         debug {
-            buildConfigField "String", "DEFAULT_BASE_URL", '"https://open-event-api-dev.herokuapp.com/v1/"'
+            buildConfigField "String", "DEFAULT_BASE_URL", '"https://api.eventyay.com/v1/"'
+
+//            buildConfigField "String", "DEFAULT_BASE_URL", '"https://open-event-api-dev.herokuapp.com/v1/"'
```
Comment (liveHarshit) [inline app/build.gradle]: revert

Comment (anhanh11001) [reply app/build.gradle]: updated

**Pull request merged**